### ISSUE TITLE
IndShockConsumerType.calc_jacobian() changes cFunc_terminal_

### ISSUE THREAD

**Issue by mnwhite** [econ-ark/HARK#1421]
The `calc_jacobian` method improperly accesses and changes `cFunc_terminal_`, which is supposed to be a class object. I'm reworking things so that `cFunc_terminal_` won't exist at all in the near future, but *instances* of a class shouldn't be changing objects defined at the class level. There is a way to do what `calc_jacobian` wants to do, but it went about it the wrong way.

This might be related to another issue we ran into lately, with a mysteriously needed duplicative call to update_solution_terminal() in order to pass tests for calc_Jacobian.

**Comment by alanlujan91** [comment]
@wdu9 @Mv77 this might be relevant to you

**Comment by Mv77** [comment]
I mean sure, its hacky.

But @wdu9 did it to get hark to do something it was not meant to do. It would be really annoying to have had to define a new class to do it properly.

The way I did it when I needed to do the same in my own branch was to add an optional extra argument to `solve` which would be a solution to serve as the start of backward induction. It makes it ignore the default terminal solution.

Should I make a PR with that change?
